### ISSUE TITLE
Implementation of Group Membership Lock

### ISSUE THREAD

**Issue by rejackson73** [gitlabhq/terraform-provider-gitlab#1133]
### Feature Description

The current GitLab provider for Terraform does not support the premium features for the Group resource, most notably the ability to "Prevent adding new members to projects within this group." The GitLab API does support this control utilizing the "membership_lock" boolean but it doesn't appear to be implemented in the provider.

### Do you want to implement this?

- [ ] I would like to implement this myself 👷

**Comment by timofurrer** [comment]
Thanks for the request!

This feature is an upstream API alignment in the `gitlab_group` resource.

* [Upstream docs](https://docs.gitlab.com/ee/api/groups.html#:~:text=or%20public.-,membership_lock,Prevent%20adding%20new%20members%20to%20projects%20within%20this%20group.,-extra_shared_runners_minutes_limit)
* `go-gitlab` already supports it.

**Comment by timofurrer** [comment]
Actually @jtymes implemented this in the GitLab REST API for the provider, see [this MR](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/82271#:~:text=This%20MR%20exposes%20the%20membership_lock%20field%20so%20this%20can%20be%20implemented%20as%20part%20of%20the%20GitLab%20Terraform%20provider.) :)

**Comment by jtymes** [comment]
Yeah it was on my list to add here, but life has been hectic so it would be good if someone else wants it.
